subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheEncryptionIntegrationTest.groovy

```diff
@@ -162,110 +137,89 @@ class ConfigurationCacheEncryptionIntegrationTest extends AbstractConfigurationC
         }
     }
 
-    def "new configuration cache entry if was encrypted but encryption is off"() {
-        given:
-        def configurationCache = newConfigurationCacheFixture()
-        runWithEncryption(true)
-
-        when:
-        runWithEncryption(false)
-
-        then:
-        configurationCache.assertStateStored()
-        outputContains("Calculating task graph as no configuration cache is available for tasks: help")
-    }
-
-    def "new configuration cache entry if was not encrypted but encryption is on"() {
+    def "new configuration cache entry if keystore is not found"() {
         given:
         def configurationCache = newConfigurationCacheFixture()
-        runWithEncryption(false)
+        runWithEncryption()
+        findKeystoreFile().delete()
 
         when:
-        runWithEncryption(true)
+        runWithEncryption()
 
         then:
         configurationCache.assertStateStored()
```

Comment:
Could add a check here to ensure the keystore was recreated at original location (assumin that is what is expected)